pynamodb/attributes.py

```diff
@@ -325,3 +326,239 @@ def deserialize(self, value):
+                    instance.attr_name = item
+                cls._attributes[item] = instance
+        return cls._attributes
+
+    def _set_attributes(self, **attrs):
+        """
+        Sets the attributes for this object
+        """
+        for attr_name, attr in self._get_attributes().aliased_attrs():
+            if attr.attr_name in attrs:
+                if type(attrs.get(attr.attr_name)) is type(attr) or not isinstance(attrs.get(attr_name), collections.Mapping):
+                    setattr(self, attr_name, attrs.get(attr.attr_name))
+                else:
+                    sub_model = attrs.get(attr_name)
+                    instance = type(attr)()
+                    aliased_attrs = type(attr)._get_attributes().aliased_attrs()
+                    for unaliased_key, v in sub_model.iteritems():
+                        k = None
+                        for python_accessible_key, aliased_v in aliased_attrs:
+                            if aliased_v.attr_name == unaliased_key:
+                                k = python_accessible_key
+                        setattr(instance, k, v)
+                    setattr(self, attr_name, instance)
+
+            elif attr_name in attrs:
+                setattr(self, attr_name, attrs.get(attr_name))
+
+    def get_values(self):
+        attributes = self._get_attributes()
+        result = {}
+        for k, v in attributes.iteritems():
+            result[k] = getattr(self, k)
+        return result
+
+    def is_type_safe(self, key, value):
+        can_be_null = value.null
+        if can_be_null and getattr(self, key) is None:
+            return True
+        return getattr(self, key) and type(getattr(self, key)) is not type(
+            value)
+
+    def validate(self):
+        for key, value in self._get_attributes().iteritems():
+            if not self.is_type_safe(key, value):
+                return False
+        return True
+
+    def serialize(self, values):
+        """
+        Encode the given list of numbers into a list of AttributeValue types.
+        """
+        rval = dict()
+        for k in values:
+            v = values[k]
+            attr_class = _get_class_for_serialize(v)
+            attr_key = _get_key_for_serialize(v)
+            if attr_class is None:
+                continue
+            if attr_key is MAP_SHORT:
+                rval[k] = [{attr_key: attr_class.serialize(v)}]
+            else:
+                rval[k] = {attr_key: attr_class.serialize(v)}
+
+        return rval
+
+    def deserialize(self, values):
+        """
+        Decode numbers from list of AttributeValue types.
+        """
+        rval = dict()
+        for k in values:
+            v = values[k]
+            attr_class = _get_class_for_deserialize(v)
+            attr_value = _get_value_for_deserialize(v)
+            rval[k] = attr_class.deserialize(attr_value)
+        return rval
+
+
+def _get_value_for_deserialize(value):
```

Comment:
is there a better way to write these functions? i want to return something different but the conditions are the same